backend/src/test/java/com/google/backend/servlets/AlertVisualizationServletTest.java

```diff
@@ -0,0 +1,108 @@
+package com.google.backend.servlets;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList; 
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.models.Alert;
+import com.google.models.Anomaly;
+import com.google.models.Timestamp;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+
+/** Contain tests for methods in {@link AlertVisualizationServlet} class. */
+@RunWith(JUnit4.class)
+public class AlertVisualizationServletTest {
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+
+  @Rule // JUnit 4 uses Rules for testing specific messages
+  public ExpectedException thrown = ExpectedException.none();
+
+  private static final String RESPONSE_CONTENT_TYPE = "application/json;";
+  private static final String ID_PARAM = "id";
+  private static final Long FAKE_ID = 1L;
+  private static final String ENTITY_NOT_FOUND_ERROR = 
+    "com.google.appengine.api.datastore.EntityNotFoundException: "
+    + "No entity was found matching the key: alert(1)";
+
```

Comment:
v nit: don't need new line here, but maybe use new line to separate `private static final` and `private final` (?)

Reply:
oh I separated them because above (including this String) are constants and below are constant objects. imo that's cleaner

Reply:
oh did not notice, sounds good!